src/main/java/org/vafer/jdeb/ControlBuilder.java

```diff
@@ -103,13 +104,19 @@ void buildControl(BinaryPackageControlFile packageControlFile, File[] controlFil
             if ("conffiles".equals(file.getName())) {
                 foundConffiles = true;
             }
-            
             if (CONFIGURATION_FILENAMES.contains(file.getName()) || MAINTAINER_SCRIPTS.contains(file.getName())) {
                 FilteredFile configurationFile = new FilteredFile(new FileInputStream(file), resolver);
                 configurationFile.setOpenToken(openReplaceToken);
                 configurationFile.setCloseToken(closeReplaceToken);
                 addControlEntry(file.getName(), configurationFile.toString(), outputStream);
 
+            } else if (file.getName().endsWith(".bin")) {
```

Comment:
If we don't use BOM and Shebang but plain heuristics we indeed run into those problems with "here documents", too.

I am wondering if we should take a step back here. Right now we have 2 types
- shell - where we apply variable expansion
- other - which we just copy without applying variable expansion

Introducing a 3rd type is a bit sneaky as we would want to have variable expansion on only the shell part of the combined file.

So question is whether there is another way to ship the jar but appending it to the shell script or having some kind of convention/marker that tells jdeb where to apply the variable expansion and what to copy as binary.